string name
string path
string metadata
